src/nestjs-winston-logger.decorator.ts

```diff
@@ -19,11 +21,15 @@ export function getLoggerToken(context: string): string {
  * Custom decorator for easy inject LoggerService
  * @param context - prefer input your service name
  */
-export function InjectLogger(context = "") {
-  loggerContexts.add(context);
+export function InjectLogger(context = "", metadata: { [key:string]: string | boolean | number } = {}) {
+  loggerContexts.set(context, metadata);
```

Comment:
There has a problem when importing the same `AUTH_LOG` of the logger inside the same module. Maybe the second will override the first according to the import sequence.
```
##Demo1Service.ts
export class Demo1Service {
  constructor(
    @InjectLogger(AUTH_LOG, { service: Demo1Service.name}) private authLogger: NestjsWinstonLoggerService) { }
}

##Demo2Service.ts
export class Demo2Service {
  constructor(
    @InjectLogger(AUTH_LOG, { service: Demo2Service.name}) private authLogger: NestjsWinstonLoggerService) { }
}
```

Reply:
Yes, since we are getting the logger through the token. 

The real situation is quite different from what I thought at my first impression (implement something like nest/mongoose). The schema remains unchanged in mongoose, while the logger may need different config in different provider...

Reply:
When I was thinking about this issue, I question popped into my head: Is there any use case that we really need to have two loggers in the same nestjs module?

E.g. The original issue, auth and request - shouldn't we use different module (or possibly another microservice?) to handle the auth process?

Reply:
I believe there are cases where you need multiple loggers in the same nestjs module.  I personally don't want everything to just fall into my application log.  I also want a separate request log in addition to the auth log.  With respect to just the specific auth case, it is probably not safe to assume that users will have their auth in a separate module (although ideally, that would be the case).

With regards to importing the same  AUTH_LOG, is your concern that logging a message in Demo1Service will produce {... "service": "Demo2Service" ...} in the case where Demo2Service gets injected after Demo1Service?  Eg, the metadata would be replaced?  I thought the transient scope would handle this, but if not, we can just drop setting the metadata from the @InjectLogger decorator.  One way to test it though could be to inject Demo1Service into Demo2Service, and log at various points during the process. I'll give it a try and let you know what I find.

Reply:
@Marcotsept  - you were right in your prediction.  The metadata got overwritten in the second injection.  I suggest we remove that option from @InjectLogger so that it only has a single parameter again.  Just taking a look a nest-winston, they allow the addition of metadata when actually logging rather than supplying it ahead of time (see ([winston.classes.ts](https://github.com/gremo/nest-winston/blob/master/src/winston.classes.ts)), the [commit](https://github.com/gremo/nest-winston/pull/228)).  A different approach that might be worth considering.

Reply:
Hi @rat-matheson , @owen1002 - Owen mentions that if the user or you wanna multiple loggers, they can import multiple modules of our dynamic module but the current version doesn't support multiple modules because the token does not identify which logger of which module.

Reply:
Oh, I understand.  For some reason I mistakenly was assuming Owen was referring to having a log and config declared in my authentication module, and not importing separate logger modules.   I should be able to try that approach.  It does seem more appropriate

Reply:
@Marcotsept @owen1002 
With regards to having separate instances of the NestjsWinstonLoggerModule, I'm wondering if it is possible to do without breaking changes.  The issue I'm having is within the forRoot, I need to be able to distinguish between a token and a context.  Previously, I was able to do this by looking at the keys in overrides as shown below.

```ts
 const loggerOptions = overrides[context] || defaultOptions;

return {
  provide: getLoggerToken(context),
  useFactory: () => {
    const logger = new NestjsWinstonLoggerService(loggerOptions);
    if (!overrides[context]) {
      logger.setContext(metadata);
    }    
    return logger;
}
```

But now I'm not able to do that because there is no master list of overrides.

Some options:

1. Break Inject Logger
```ts @InjectLogger(logToken:string, context?:string)``` - in this case, logToken is always required which makes it easy to create a token that incorporates it. It breaks the case @InjectLogger(SomeClass.name)

2. Ditch using context when injected and instead apply it when logging

3. Have some construct that makes it possible to determine if the first parameter is a token or context
@InjectLogger(logTokenOrContext:string, context?:string) - in this case, it is assumed the first param is context unless there are 2 params
@InjectLogger(logTokenOrContext:string | { token:string } )


I have other ideas but they keep getting worse so I'll stop here.

Reply:
@Marcotsept @owen1002 - do you guys have any ideas with regards to the breaking change issues I encountered?